### ISSUE TITLE
smartAuth.requestHint getting crash 

### ISSUE THREAD

**Issue by Connect-Suraj-Chaudhari** [Tkko/flutter_smart_auth#22]
Getting crash when requestinh phone hint

 java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/android/gms/auth/api/credentials/HintRequest$Builder;

Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.android.gms.auth.api.credentials.HintRequest$Builder" on path: DexPathList[[zip file "/data/app/~~b646lIclmArAMB3At_-jyw==/com.app-Lfbja4qsyhQwiTlaNeLhiA==/base.apk"],nativeLibraryDirectories=[/data/app/~~b646lIclmArAMB3At_-jyw==/com.app-Lfbja4qsyhQwiTlaNeLhiA==/lib/arm64, /data/app/~~b646lIclmArAMB3At_-jyw==/com.app-Lfbja4qsyhQwiTlaNeLhiA==/base.apk!/lib/arm64-v8a, /system/lib64, /system/system_ext/lib64]]

**Comment by Shub0327** [comment]
same issue, found any solution?

**Comment by amitkumarmidya1987** [comment]
Getting crash when smartAuth.requestHint email hint

**Crash report**
Fatal Exception: java.lang.NoClassDefFoundError
Failed resolution of: Lcom/google/android/gms/auth/api/credentials/HintRequest$Builder;
man.ge.smart_auth.SmartAuthPlugin.requestHint (SmartAuthPlugin.kt:116)
fman.ge.smart_auth.SmartAuthPlugin.onMethodCall (SmartAuthPlugin.kt:76)
io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage (MethodChannel.java:258)
io.flutter.embedding.engine.dart.DartMessenger.invokeHandler (DartMessenger.java:295)
io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0 (DartMessenger.java:322)
android.os.Handler.handleCallback (Handler.java:883)

**Comment by DurgeshSawantWYLD** [comment]
https://developers.google.com/android/guides/releases#february_15_2024
as per the official docs they are deprecated the credentials API - please update the imports thanks!

**Comment by Shub0327** [comment]
Hey @DurgeshSawantWYLD, can you share details of what changes you did in package to resolve this thing?

**Comment by Astaxask** [comment]
Please author, we need an update!

**Comment by irshad-dotcod** [comment]
Getting crash when requestinh phone hint

java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/android/gms/auth/api/credentials/HintRequest$Builder;

Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.android.gms.auth.api.credentials.HintRequest$Builder" on path: DexPathList[[zip file "/data/app/~~b646lIclmArAMB3At_-jyw==/com.app-Lfbja4qsyhQwiTlaNeLhiA==/base.apk"],nativeLibraryDirectories=[/data/app/~~b646lIclmArAMB3At_-jyw==/com.app-Lfbja4qsyhQwiTlaNeLhiA==/lib/arm64, /data/app/~~b646lIclmArAMB3At_-jyw==/com.app-Lfbja4qsyhQwiTlaNeLhiA==/base.apk!/lib/arm64-v8a, /system/lib64, /system/system_ext/lib64]]

**Comment by dishankjindal1** [comment]
E/AndroidRuntime(15361): FATAL EXCEPTION: main
E/AndroidRuntime(15361): Process: com.pennyfintech.sikka.stag, PID: 15361
E/AndroidRuntime(15361): java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/android/gms/auth/api/credentials/HintRequest$Builder;
E/AndroidRuntime(15361): 	at fman.ge.smart_auth.SmartAuthPlugin.requestHint(SmartAuthPlugin.kt:116)
E/AndroidRuntime(15361): 	at fman.ge.smart_auth.SmartAuthPlugin.onMethodCall(SmartAuthPlugin.kt:76)
E/AndroidRuntime(15361): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:267)
E/AndroidRuntime(15361): 	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:292)
E/AndroidRuntime(15361): 	at io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0$io-flutter-embedding-engine-dart-DartMessenger(DartMessenger.java:319)
E/AndroidRuntime(15361): 	at io.flutter.embedding.engine.dart.DartMessenger$$ExternalSyntheticLambda0.run(Unknown Source:12)
E/AndroidRuntime(15361): 	at android.os.Handler.handleCallback(Handler.java:958)
E/AndroidRuntime(15361): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime(15361): 	at android.os.Looper.loopOnce(Looper.java:255)
E/AndroidRuntime(15361): 	at android.os.Looper.loop(Looper.java:364)
E/AndroidRuntime(15361): 	at android.app.ActivityThread.main(ActivityThread.java:8938)
E/AndroidRuntime(15361): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(15361): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:572)
E/AndroidRuntime(15361): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1053)
E/AndroidRuntime(15361): Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.android.gms.auth.api.credentials.HintRequest$Builder" on path: DexPathList[[zip file "/data/app/~~nyzqZi0kGk1QFYaoK9kUiw==/com.pennyfintech.sikka.stag-7HZgKTJgSx5qXR9yiOcRew==/base.apk"],nativeLibraryDirectories=[/data/app/~~nyzqZi0kGk1QFYaoK9kUiw==/com.pennyfintech.sikka.stag-7HZgKTJgSx5qXR9yiOcRew==/lib/arm64, /data/app/~~nyzqZi0kGk1QFYaoK9kUiw==/com.pennyfintech.sikka.stag-7HZgKTJgSx5qXR9yiOcRew==/base.apk!/lib/arm64-v8a, /system/lib64, /system_ext/lib64]]
E/AndroidRuntime(15361): 	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:259)
E/AndroidRuntime(15361): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
E/AndroidRuntime(15361): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
E/AndroidRuntime(15361): 	... 14 more
I/Process (15361): Sending signal. PID: 15361 SIG: 9

**Comment by TechGeekD** [comment]
I am not a Kotlin expert but I've managed to fix requestHint crash for the phone number ->

```  
smart_auth:
    git:
      url: "https://github.com/metaspace-src/flutter_smart_auth"
      ref: "main"
```

- Support for email is not available in requestHint as I've just added fix for the phone number

- [ ] Issue: the phone number is being picked with the country code picked from the phone language setting.
i.e. If you are from India & set your phone language as English(United Kingdom) then you will get +44 along with your phone number

**Comment by nikb7** [comment]
https://pub.dev/packages/phone_hint_android

You can try this lib to get the phone number

**Comment by Tkko** [comment]
Hey guys, The fix for this issue is already on the main branch but it isn't yet uploaded to the pub.dev
If you need to fix it ASAP, you can depend on the directly main branch:

```
  smart_auth:
    git:
      url: "git@github.com:Tkko/flutter_smart_auth.git"
      ref: fc1c5de
```
See readme and the [change log](https://github.com/Tkko/flutter_smart_auth/blob/main/CHANGELOG.md) for more details.

**Comment by Tkko** [comment]
Fixed in version [3.0.0](https://pub.dev/packages/smart_auth)
